### PR TITLE
Migrate org.elasticsearch.action to TransportVersion

### PR DESCRIPTION
**Pull request by thecoop**
Migrate files in `org.elasticsearch.action` to use TransportVersion.

Vast majority is `s/Version/TransportVersion/`, with a couple of comments

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java [hunk 1]
```diff
@@ -235,7 +235,7 @@ protected void doInternalExecute(Task task, BulkRequest bulkRequest, String exec
             }
 
             if (actionRequest instanceof IndexRequest ir) {
-                ir.checkAutoIdWithOpTypeCreateSupportedByVersion(minNodeVersion);
+                ir.checkAutoIdWithOpTypeCreateSupportedByVersion(minNodeVersion.transportVersion);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java:238]: `minNodeVersion` will need to be migrated separately, its used in several other places

File: server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java [hunk 2]
```diff
@@ -158,40 +158,44 @@ public void writeTo(StreamOutput out) throws IOException {
...
         out.writeBoolean(verbose);
-        if (out.getVersion().onOrAfter(PAGINATED_GET_SNAPSHOTS_VERSION)) {
+        if (out.getTransportVersion().onOrAfter(PAGINATED_GET_SNAPSHOTS_VERSION)) {
             out.writeOptionalWriteable(after);
             if ((sort == SortBy.SHARDS || sort == SortBy.FAILED_SHARDS || sort == SortBy.REPOSITORY)
-                && out.getVersion().before(SORT_BY_SHARDS_OR_REPO_VERSION)) {
+                && out.getTransportVersion().before(SORT_BY_SHARDS_OR_REPO_VERSION)) {
                 throw new IllegalArgumentException(
-                    "can't use sort by shard count or repository name with node version [" + out.getVersion() + "]"
+                    "can't use sort by shard count or repository name with version [" + out.getTransportVersion() + "]"
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java:166]: perhaps node->transport in the exception message would be useful?

File: server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java [hunk 3]
```diff
@@ -158,40 +158,44 @@ public void writeTo(StreamOutput out) throws IOException {
...
             out.writeEnum(sort);
             out.writeVInt(size);
             order.writeTo(out);
-            if (out.getVersion().onOrAfter(NUMERIC_PAGINATION_VERSION)) {
+            if (out.getTransportVersion().onOrAfter(NUMERIC_PAGINATION_VERSION)) {
                 out.writeVInt(offset);
             } else if (offset != 0) {
                 throw new IllegalArgumentException(
-                    "can't use numeric offset in get snapshots request with node version [" + out.getVersion() + "]"
+                    "can't use numeric offset in get snapshots request with version [" + out.getTransportVersion() + "]"
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java:176]: node->transport version might be more clear?

File: server/src/main/java/org/elasticsearch/action/search/SearchRequest.java [hunk 4]
```diff
@@ -303,27 +304,27 @@ public void writeTo(StreamOutput out) throws IOException {
...
-        if (out.getVersion().onOrAfter(waitForCheckpointsVersion)) {
+        TransportVersion waitForCheckpointsVersion = TransportVersion.V_7_16_0;
+        if (out.getTransportVersion().onOrAfter(waitForCheckpointsVersion)) {
             out.writeMap(waitForCheckpoints, StreamOutput::writeString, StreamOutput::writeLongArray);
             out.writeTimeValue(waitForCheckpointsTimeout);
         } else if (waitForCheckpoints.isEmpty() == false) {
             throw new IllegalArgumentException(
                 "Remote node version ["
-                    + out.getVersion()
+                    + out.getTransportVersion()
```
[the remark this answers is not in the document]
Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/action/search/SearchRequest.java:320]: I think it should be transportVersion.

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-2

**Review by pgomulka — APPROVED** [review]
LGTM

**Pull request merged**